### PR TITLE
Rename element ref coords integer coords

### PR DESCRIPTION
**Pull request by sandro-elsweijer**
**_Describe your changes here:_**
I was heavily confused by the current naming of the element reference coords functions, so I renamed the integer versions

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Comment by lukasdreyer** [comment]
This function is not actually part of the c interface, and I would argue that it should not be, since it assumes that the SFC internally uses integer coordinates on `[0, 2^l)`, which the SFC does not need to do. The internal counterpart can be used for the version that is scaled to [0,1], but should not be accessible from the outside

**Comment by lukasdreyer** [comment]
@holke: I had a discussion offline with @sandro-elsweijer about the necessity of this functionality in the interface, and he mentioned that you had some arguments why it should not be removed. Could you elaborate on that?
In my opinion, this functionality as well as the `anchor` functionality are implementation details, if the user wants to access geometry information of an element, they should use `t8_element_reference_coords` or `t8_element_vertex_reference_coords`

File: src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx [hunk 1]
```diff
@@ -674,19 +674,17 @@ t8_default_scheme_quad_c::t8_element_anchor (const t8_element_t *elem, int coord
 }
 
 void
-t8_default_scheme_quad_c::t8_element_vertex_coords (const t8_element_t *elem, int vertex, int coords[]) const
+t8_dquad_vertex_integer_coords (const p4est_quadrant_t *elem, int vertex, int coords[])
 {
-  const p4est_quadrant_t *q1 = (const p4est_quadrant_t *) elem;
   int len;
```
Comment (Niklas997) [inline src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx:679]:
```suggestion
```

File: src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx [hunk 2]
```diff
@@ -674,19 +674,17 @@ t8_default_scheme_quad_c::t8_element_anchor (const t8_element_t *elem, int coord
...
 
-  T8_ASSERT (t8_element_is_valid (elem));
   T8_ASSERT (0 <= vertex && vertex < 4);
   /* Get the length of the quadrant */
-  len = P4EST_QUADRANT_LEN (q1->level);
+  len = P4EST_QUADRANT_LEN (elem->level);
```
Comment (Niklas997) [inline src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx:683]:
```suggestion
  const int len = P4EST_QUADRANT_LEN (elem->level);
```

File: src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx [hunk 3]
```diff
@@ -541,21 +541,19 @@ t8_default_scheme_hex_c::t8_element_anchor (const t8_element_t *elem, int coord[
   coord[2] = q->z;
 }
 
-void
-t8_default_scheme_hex_c::t8_element_vertex_coords (const t8_element_t *elem, int vertex, int coords[]) const
+static void
+t8_dhex_vertex_integer_coords (const p8est_quadrant_t *elem, int vertex, int coords[])
 {
-  const p8est_quadrant_t *q1 = (const p8est_quadrant_t *) elem;
   int len;
```
Comment (Niklas997) [inline src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx:547]:
```suggestion
```

File: src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx [hunk 4]
```diff
@@ -541,21 +541,19 @@ t8_default_scheme_hex_c::t8_element_anchor (const t8_element_t *elem, int coord[
...
 
-  T8_ASSERT (t8_element_is_valid (elem));
   T8_ASSERT (0 <= vertex && vertex < 8);
   /* Get the length of the quadrant */
-  len = P8EST_QUADRANT_LEN (q1->level);
+  len = P8EST_QUADRANT_LEN (elem->level);
```
Comment (Niklas997) [inline src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx:551]:
```suggestion
  const int len = P8EST_QUADRANT_LEN (elem->level);
```

**Review by Niklas997 — CHANGES_REQUESTED** [review]
The pull request looks good! Please merge the current main branch and resolve the conflicts. Afterwards, we are able to merge it.

**Review by Niklas997 — APPROVED** [review]

**Pull request merged**